Versão 1.4: Melhorias na pesquisa por atributos

sabrin477 opened the pull request:
Problema: erros do back-end não são interpretados pelo front-end

paschendale committed:
Correções de responsividade do menu e do container de legendas

paschendale committed:
Atualização de Todo's

paschendale committed:
Sistema de referencia definido em configurações

O Model de Config agora tem a permissão de alterar a tabela, isto é, a criação de novas colunas no model implica na criação dela também na base mesmo após a tabela já ter sido criada.

Este tipo de configuração não é o recomendado para produção, mas vamos seguindo com ele até desenvolvermos algum método mais adequado.

Detalhe: apenas o model Config possui essa configuração. Users e Layers não, acredito que não seja proveitoso termos também em razão de segurança das bases já existentes durante atualizações do sistema.

Detalhe 2: O default para o sistema de referência é o EPSG 31983, ou seja, UTM 23 S SIRGAS 2000.

paschendale committed:
Seleção de feições via Point in Polygon

Empregada solução ECQL + CQL

sabrin477 committed:
Removendo Formidable

sabrin477 committed:
Revert "Removendo Formidable"

This reverts commit fe4fc2486469139be717369fbd654b400f119ede.

sabrin477 committed:
Revert "Revert "Removendo Formidable""

This reverts commit 0b8076ba1b16528b0a5b62fc3e82af317d512336.

sabrin477 committed:
Upload de arquivos metadados

paschendale committed:
Bug fix para lançamento da 1.1

paschendale committed:
Merge branch 'main' into dev

paschendale committed:
Bug fix Metadados indefinidos

paschendale committed:
Bug fix - Formidable

sabrin477 committed:
Botão de remover metadata

paschendale committed:
Tooltips, Dark mode, remoção da navbar e alteração do Host default

- Host default alterado de mapas.genteufv para maps.genteufv, para compatibilizar com nosso novo geoserver de demonstração. Possivelmente será alterado posteriormente. Credenciais padrão não funcionam no mapas.genteufv por questões de segurança, nos próximos commits isto será corrigido

- Adicionado dark mode automatico toda vez que a color scheme do navegador ou sistema for escura. Todas as alterações de estilo são listadas após o comentário 'Dark mode' em webgente.css

- Alterada attribution do sistema de Leaflet para WebGENTE

- Removida a navbar, funcionalidades de login e help transferidas para botões

- Botões de ferramentas agrupados utilizando o easyBar, uma função do easyButton

- Traduzidos os textos da ferramenta de desenhos

- O js do Leaflet.Draw agora está local

- Adicionados tooltips para todas as ferramentas

- Requisicoes de GetLegendGraphics agora vêm transparentes

guirosemberg committed:
Ícones para a barra lateral

paschendale committed:
Habilitando/desabilitando ferramentas, Legendas Customizadas, Melhoria da Interface do Usuário e outras alterações

### Habilitando e desabilitando ferramentas

- Adicionada interface de habilitar/desabilitar ferramentas em config_tools.ejs

- Adicionados campos booleanos em configurações para habilitar/desabilitar ferramentas

Até o momento foram criadas as variáveis:

home_enabled -> Desabilita o botão de volta à vista inicial
select_enabled -> Desabilita a ferramenta de seleção de feições
information_enabled -> Desabilita a ferramenta de visualização de atributos de feições
search_enabled -> Desabilita a ferramenta de pesquisa por atributo
legend_enabled -> Desabilita a ferramenta de legendas
geolocation_enabled -> Desabilita a geolocalização
measurement_enabled -> Desabilita as ferramentas de medição
custom_legend_enabled -> Desabilita as legendas customizadas
coordinates_enabled -> Desabilita a ferramenta de coordenadas

- Novas ferramentas são desabilitadas por default na atualização do sistema, devendo ser habilitadas em configurações

- A funcionalidade permite habilitarmos ferramentas em modo 'beta' para irmos testando na main sem comprometer a atualização de sistemas em produção

- Neste commit são desabilitadas as funcionalidades: pesquisa por atributos e legenda personalizada, para habilitar por padrão deve-se alterar o valor da variavel inserida em dummyData

- Ferramentas sao desabilitadas somente visualmente na interface através de webgente.js, código para a funcionalidade está e deve permanecer nas ultimas linhas

### Legendas customizadas

- Almejando a melhoria do dark mode futuramente agora é possível utilizar legendas customizadas pelo Geoserver. A funcionalidade remove o título da legenda ao acionar a camada e insere um <hr> entre as imagens com um aspecto mais minimalista.

### Melhoria da interface do usuário

- Removido o atributo birthDate das interfaces do usuário

- Password agora é editável na interface de edição

- Grupo do usuário é selecionado por menu

### Outras alterações

- Servidor de testes para o WebGENTE retornado para o mapas.genteufv.com.br

- Adicionado HTML e CSS para criação do teste de usuário/senha/endereço do servidor em Config. Irei implementar a funcionalidade via backend pois via front não deu certo

- Adicionado campo para alteração da attribution das camadas base e MDT em layer_details.ejs, agora é possível alterar o texto que aparece no canto inferior direito do mapa, inclusive com HTML

paschendale committed:
Ocultando o webgente-legend-container quando desnecessário

- a legenda só aparece agora quando duas condições forem atingidas:

1. existe alguma camada ligada
2. o atributo data-state de webgente-legend-container for igual à 'enabled'

- o botão de ativação/desativação das legendas agora apenas altera o data-state

paschendale committed:
Alteração de host da base de demonstração

- Removida coluna birthDate de users.ejs

- Alterado o host e workspace de demonstração do WebGENTE, adicionado um modelo digital de elevação

sabrin477 committed:
Alteração no diretório do upload

-- Alteração do caminho de barras invertidas para uma barra apenas, evitando erros de upload nos sistemas linux.

paschendale committed:
Merge branch 'main' into dev

sabrin477 committed:
Restrição na ferramenta de seleção

-- Alteração nos resultados da rota Select: propriedades inclusas com restrição de tributos;
-- Remoção da função removeProperties  na rota Select;
--Restrição de campos na rota Select utilizando a função restrictAttributes.

paschendale committed:
Bug fix - Erro na criação de usuários

Como birthDate não é mais inserido a constraint allowNull: false estava dando erro nas inserções.

Adicionado defaultValue na coluna birthDate no model de Users e adicionado alter: true.

paschendale committed:
Feed de novidades do WebGENTE no home da interface administrativa

O feed é alimentado através do RSS da tag WebGENTE do Blog: https://genteufv.com.br/blog/index.php/category/webgente/feed/

O RSS é lido por uma API online ( https://api.rss2json.com/v1/api.json?) e transformado em JSON. A API atualiza o feed diariamente apenas.

O JSON é lido no frontend, onde são construídos os cards do Feed através da função feedCardHTML()

São exibidos somente os 4 posts mais recentes da tag!

sabrin477 committed:
Habilitando/desabilitando download da ferramenta de pesquisa

-- O download pode ser habilitado somente para usuários logados
-- Habilitar/Desabilitar download está vinculado ao checkbox do ferramenta de pesquisa.

sabrin477 committed:
Correção: Habilitando/Desabilitando download

-- Download para  TODOS os usuários
--Segurança adicional no backend na rota wfs

paschendale committed:
Adicionado suporte ao reordenamento de camadas (#5)

* Atualização com o Roadmap no README.md

* Interface para reordenamento de camadas

- Adicionado botão na área de administração de camadas para realizar o reordenamento desta

- Camadas são sobrepostas conforme sua ordem de chamada, sendo assim, camadas mais abaixo do menu de camadas sobrepõem aquelas mais acima. Isto estava criando alguns efeitos indesejados à depender da ordem de declaração das camadas na interface: uma camada de quadra declarada após uma camada de lotes impediria a visualização de ambas camadas simultaneamente devido à sobreposição da primeira à segunda.

- A interface de reordenamento permite reordenar tanto os grupos quanto as camadas dentro dos grupos

- O reordenamento é feito via uma requisição POST à rota /layers/reorder, no corpo da requisição são enviados os id's da ordem antiga das camadas e a nova ordem, realiza-se o reordenamento alterando os id's, e, dessa forma, alterando a forma como o Sequelize serve as camadas por todas as interfaces (ordenadas pelo novo id)

- A criação da lista é realizada dinamicamente e o draggable é dado pelo jQuery-UI, colocado via CDN unicamente na página reorder.ejs

- Para todos os elementos foram atribuidas classes e linkadas em administracao.css, ainda que nao tenham personalizações. O CSS principal neste commit é todo do Bootstrap

paschendale committed:
Bug fix - Decoding de URIs em proxys reversos

COMMIT TESTE

Quando empregando proxys reversos no WebGENTE é necessário adicionar uma diretiva AllowEncodedSlashes setada para Decode para permitir que as requisições url-codificadas no frontend cheguem ao backend corretamente, ou seja, sem decodificar os slashes (/ -> %2F)

Essa diretiva faz com que alguns parametros cheguem codificados ao backend e não são decodificados, por ex. o outputFormat = 'application/json' que é enviado ao geoserver como 'outputFormat=application%252Fjson'

Para resolver este problema criei uma função decodeURIComponentSafely ((Problema parecido no StackOverflow)[https://stackoverflow.com/questions/7449588/why-does-decodeuricomponent-lock-up-my-browser/67905828#67905828])

Basta envolver os parametros recebidos em req.params com decodeURIComponentSafely que tudo deve resolver

paschendale committed:
Partial Revert "Bug fix - Decoding de URIs em proxys reversos"

This reverts partially commit ccddb5218792900cdeab980f19928f0d81f1c3fe.

paschendale committed:
Adicionado decode para URL do Host

Solução para compatibilizar o uso do proxy reverso

paschendale committed:
Adicionado decode para URL do Host

This reverts commit 71764a75aa6cabb0ab8615c6c676d5cdd46535f0.

paschendale committed:
DescribeLayer - removida verificação se host é URL

DescribeLayer - removida verificação se host é URL

paschendale committed:
DescribeLayer - removida verificação se host é URL

DescribeLayer - removida verificação se host é URL

paschendale committed:
Adicionada pagina de Help

- Página de Help na interface de Administração exibirá vídeos do Feed da playlist de Tutoriais da Interface de Administração

- Mecanismo de atualização do feed é o mesmo do de atualização do blog, empregando a API do rss2json

- Futuramente trocaremos esse feed pelo feed da documentação

sabrin477 committed:
Solução: Erro de arquivos exportados na pesquisa

-- Alteração do nome do arquivo exportado
-- Mudança de formatos exportados: GML para XML
-- Alteração no método de exportar feições únicas.

sabrin477 committed:
Revert "Solução: Erro de arquivos exportados na pesquisa"

This reverts commit f851f23fa6041bd6684cb5b55c6c3bee627c8853.

paschendale committed:
Merge branch 'main' into dev

paschendale committed:
Exclusao de node_modules + Partial Revert "Merge branch 'main' into dev"

This reverts commit b2496aad6285d23024e87eebe5f36ca68be0e227, reversing
changes made to 9d68f4f656250b0c9ec0d4072a89922f90dbf81f.

sabrin477 committed:
Correção: formatos de exportação e spinners

- Alteração do formato gml para kml
-Nomes dos arquivos exportados foram alterados de "feicao" para o nome da camada
- Mudança na exportação do geojson: são realizadas requisições ao backend
-Adição da variável opcional srs_name a rota wfs para dados que devem possuir mesmo sistema que o utilizado no leaflet.
-Adição de spinners aos botões durante a requisição de dados.

sabrin477 committed:
Correção de Erros

- Correção de erros para pesquisa vazia: retorna todos os objetos da camada ( cql_filter removido da requisição)
-Correção de erros no back-end: retorna status de erro e confere possíveis strings de erro
-Correção de erros front-end: confere possíveis json de erro e alert para erros no ajax

paschendale reviewed (changes requested):
Download de todas as feições pesquisadas
- Arquivo CSV sendo gerado com extensão .cvs
- Todos os formatos estão sendo gerados com 0 bytes

Download de feição individual
- Arquivo está sendo gerado com 0 bytes

sabrin477 committed:
Erros download

- Correção de formato csv
- Correção de downloads vazios adicionando retorno dos resultados

paschendale committed:
Merge branch 'main' into pesquisa-atributos
